test/basic-querying.test.js

```diff
@@ -480,12 +481,23 @@ describe('basic-querying', function() {
       });
 
     it('should support string "gt" that is not satisfied', function(done) {
-      User.find({where: {name: {'gt': 'xyz'},
-      }}, function(err, users) {
-        should.not.exist(err);
-        users.should.have.property('length', 0);
-        done();
-      });
+      // IBM i tables are coded in EBCDIC by default, need to modify the query
+      // so that it returns no results
+      if (db.adapter.name === 'ibmi') {
+        User.find({where: {name: {'gt': 'XYZ'},
```

Comment:
@markdirish , any reason why we need to test for 'XYZ' as well as 'xyz' for 'gt'?

The tests for 'lt' don't test for all lower and then all upper case. Just wondering

Reply:
As mentioned in the comment above the test (although it is cryptic looking at it with new eyes), the tables that are created in the Db2 for i DBMS are encoded in EBCDIC by default. In EBCDIC, capital letters come after lowercase letters (have a greater value)

In the tables this test is using, the names begin with a capital, like "Mark". So in ASCII, where capitals come first, the test passes fine (as 'M' has a decimal value of 77, and 'x' has a value of 120, and we are looking for strings > 'xyz'). But in EBCDIC the opposite is true ('M' has a value 212, 'x' has a value of 167), so the `User.find` will return results, and then the test fails (because it expects _no_ results).

The `lt` test as written passes in either scenario, because it is comparing capital letters to capital letters, and A has a lower value than Z in both EBCDIC and ASCII

Reply:
Uh oh, this is tricky. 

> the tables that are created in the Db2 for i DBMS are encoded in EBCDIC by default.

Would it make sense to configure the database (or at least the `User` table) to use ASCII-based encoding instead of the default EBCDIC? I am imagining that with that change we won't need to modify test code. The downside is that we will no longer test integration of LoopBack with EBCDIC-encoded tables - is that a problem? What encoding is typically used by Db2 for i DBMS customers?

If we decide it's better to test against EBCDIC, then I would like the code comment to capture the detailed explanation from https://github.com/strongloop/loopback-datasource-juggler/pull/1810/files#r356624276.

Reply:
I'd say 99% or greater of IBM i users keep the standard EBCDIC encoding. It has implications for all of the code written over the last 30+ years, so it would be hard to get them to switch now.

I can update the code comment to be a little more descriptive

Reply:
> I'd say 99% or greater of IBM i users keep the standard EBCDIC encoding. It has implications for all of the code written over the last 30+ years, so it would be hard to get them to switch now.

Makes sense.

What if we changed the value compared against from letters to numbers, would that allow us to use the same query for both ASCI and EBCDIC encoding?

If not, then I am fine to accept your current proposal as long as the code comment is updated to be more descriptive.

Reply:
Nope! In ASCII, numbers precede all letters, while in EBCDIC, number are after all letters. Looks like there is no set of character that would work for a "gt" that is not satisfied. On the more recent commit, I believe I have a much more descriptive comment (and reduced code duplication)